packages/headless/src/features/result-preview/result-preview-slice.ts

```diff
@@ -19,6 +20,14 @@ export const resultPreviewReducer = createReducer(
       .addCase(updateContentURL.fulfilled, (state, action) => {
         const {contentURL} = action.payload;
 
+        state.contentURL = contentURL;
+      })
+      .addCase(executeSearch.fulfilled, (state) => {
+        const {content, isLoading, uniqueId, contentURL} =
+          getResultPreviewInitialState();
+        state.content = content;
+        state.isLoading = isLoading;
+        state.uniqueId = uniqueId;
         state.contentURL = contentURL;
```

Comment:
I think destructuring would be more readable here (suggestion WIP, prettier incoming)

```suggestion
      .addCase(executeSearch.fulfilled, (state) => {
        ({content: state.content, isLoading: state.isLoading, uniqueId: state.uniqueId, contentURL: state.contentUrl} =
          getResultPreviewInitialState());
```

Reply:
With hindsight: Does not work due to immutability enforced by redux tool kit.